### PR TITLE
[concepts] Replace 'could' and 'might'

### PR DESCRIPTION
**Pull request by jensmaurer**
as directed by ISO/CS.

Partially addresses #4319

### CODE REVIEW

File: source/concepts.tex [hunk 1]
```diff
@@ -1138,8 +1138,8 @@
 \pnum
 \begin{note}
 The \libconcept{semiregular} concept is modeled by types that behave similarly
-to built-in types like \tcode{int}, except that they might not
-be comparable with \tcode{==}.
+to built-in types like \tcode{int}, except that they are not necessarily
```
Comment (CaseyCarter) [inline source/concepts.tex:1141]: I think I'd prefer "they need not be comparable" to "they are not necessarily comparable" here, but it's not a huge issue.

**Review by CaseyCarter — APPROVED** [review]

File: source/concepts.tex (hunk 1 above)
Comment (jensmaurer) [reply source/concepts.tex]: fixed + force-pushed

**Review by CaseyCarter — APPROVED** [review]
Approved harder!

File: source/concepts.tex [hunk 2]
```diff
@@ -394,7 +394,7 @@
 are modeled, then \tcode{T} and \tcode{U} share a
 \term{common reference type}, \tcode{C}.
 \begin{note}
-\tcode{C} could be the same as \tcode{T}, or \tcode{U}, or it could be a
+\tcode{C} can be the same as \tcode{T}, or \tcode{U}, or it can be a
```
Comment (zygoloid) [inline source/concepts.tex:397]:
```suggestion
\tcode{C} can be the same as \tcode{T} or \tcode{U}, or can be a
```

File: source/concepts.tex [hunk 3]
```diff
@@ -437,8 +437,8 @@
 \tcode{C}, then \tcode{T} and \tcode{U} share a \term{common type},
 \tcode{C}.
 \begin{note}
-\tcode{C} could be the same as \tcode{T}, or \tcode{U}, or it could be a
-different type. \tcode{C} might not be unique.
+\tcode{C} can be the same as \tcode{T}, or \tcode{U}, or it can be a
```
Comment (zygoloid) [inline source/concepts.tex:440]:
```suggestion
\tcode{C} can be the same as \tcode{T} or \tcode{U}, or can be a
```

**Pull request merged**